app/src/main/java/uz/direction/coroutineshomework/MemesService.kt

```diff
@@ -1,46 +1,38 @@
 package uz.direction.coroutineshomework
 
 import android.content.Context
+import android.util.Log
 import com.google.gson.Gson
-import com.google.gson.JsonSyntaxException
-import java.io.IOException
+import kotlinx.coroutines.*
 import java.io.InputStream
 import java.nio.charset.Charset
-import kotlin.concurrent.thread
 import kotlin.random.Random
 
-class MemesService {
+val memeScope =
+    CoroutineScope(
+        Dispatchers.Main +
+                SupervisorJob() +
+                CoroutineExceptionHandler { _, exeption ->
+                    Log.e("Exection", "Exception $exeption is caught")
+                }
+                + CoroutineName("MainCoroutine"))
 
-    fun subscribeToMemes(memesListener: MemesListener, context: Context) {
-        thread {
-            val jsonString: String = try {
-                val inputStream: InputStream = context.assets.open("facts.json")
-                val size: Int = inputStream.available()
-                val buffer = ByteArray(size)
-                inputStream.read(buffer)
-                inputStream.close()
-                String(buffer, Charset.defaultCharset())
-            } catch (e: IOException) {
-                memesListener.onError(e)
-                ""
-            }
+class MemesService {
 
-            try {
-                val memes = Gson().fromJson(jsonString, Array<Meme>::class.java)
-                val randomIndex = Random.nextInt(memes.size - 1)
-                memesListener.onSuccess(memes[randomIndex])
-            } catch (e: JsonSyntaxException) {
-                memesListener.onError(e)
-            }
+    suspend fun subscribeToMemes(context: Context): Meme {
+        val job = memeScope.async {
```

Comment:
You don't need one more scope here, it is already suspend function that will definitely run in scope. What should you do is change context of a scope to IO